docs/conventions/component.md

```diff
@@ -0,0 +1,527 @@
+@include check-imports(('ec-component-link', 'ec-component-button'));
+
+// Use generic mixin
+@include exports('ec-component-navigation-menu') {
+  @include ecl-navigation-menu(
+    $bar-bg-mobile: map-get($ecl-colors, 'grey-15'),
+    $bar-bg-desktop: map-get($ecl-colors, 'grey-10'),
+    $toggle-btn-color: map-get($ecl-colors, 'blue-100'),
+    $group-bg: map-get($ecl-colors, 'grey-5'),
+    $separator-mobile: map-get($ecl-colors, 'grey-15'),
+    $link-color: map-get($ecl-colors, 'blue-100'),
+    $active-color: map-get($ecl-colors, 'blue-100'),
+    $active-bg: map-get($ecl-colors, 'grey-10')
+  );
+
+  // Add custom rules
+  .ecl-navigation-menu__item--active {
+    [...]
+  }
+  .ecl-navigation-menu__group::before {
+    [...]
+  }
+}
+```
+
+### Specific case: editor preset
+
+If the component alters an element corresponding to a standard HTML tag (like links alters <a\>, blockquotes alters <blockquote\>, ...), it should also provide CSS rules for editing.
+
+**Please note that this is a specific case, most components don't have to create editor rules**
+
+To add editor rules, there should be some extra rules (prefixed by `.ecl-editor`), which override default HTML tag.
+This rules should be put in a mixin called `ecl-editor-{component name}`, in a separated file called `{component name}--editor.scss`.
+
+Example:
+_(File name: generic-component-link--editor.scss)_
+
+```scss
+@mixin ecl-editor-generic-component-link() {
+  .ecl-editor a {
+    box-sizing: border-box;
+    color: $ecl-color-primary;
+    margin: 0;
+    text-decoration: underline;
+    [...]
+  }
+}
+```
+
+## Twig
+
+Every component should have a directly usable Twig file. The goal is to be able to include the code in other components or templates (by putting most of the content in context), without having to duplicate it.
+
+Apart from [rules that should be applied to all Twig files](twig), there are some specifications for components:
+
+### Generic component
+
+Generic component Twig file defines all the markup for the component. It should contain all the logic required to use the component, as it is intended to be used by systems components (or anything based on Twig).
+
+This Twig file consists of different sections (see link above for more details):
+
+* Parameters/Blocks: list of exposed parameters and blocks
+* Internal properties: definition of custom variables
+* Internal logic: preprocess and other custom logic
+* Print: HTML markup
+
+Example:
+_(File name: generic-component-blockquote.twig)_
+
+```twig
+{#
+  Parameters:
+    - "body" (string) (default: '')
+    - "author" (string) (default: '')
+    - "extra_classes" (string) (default: '')
+    - "extra_attributes" (array) (default: ''): format: [{ 'name': 'name_of_the_attribute', 'value': 'value_of_the_attribute'}])
+#}
+
+{# Internal properties #}
+
+{% set _body = body|default('') %}
+{% set _author = author|default('') %}
+{% set _css_class = 'ecl-blockquote' %}
```

Comment:
`_css_class` vs `_css_classes`?

Reply:
I can accept both, but we have to take a decision before updating all twig files